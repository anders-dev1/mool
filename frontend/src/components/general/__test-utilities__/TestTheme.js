export const testTheme = {
    textColor: "black",
    primaryColor: "#007BFF",
    primaryHover: "rgba(0,123,255,0.33)",
    successColor: "#28A745",
    errorColor: "#F02849",
    backgroundColor: "#F0F2F5",
    buttonTextColor: "#FFFFFF",
    dividerColor: "#C7CBD1",
    inputColor: "#99A5B0",
    spinnerColor: "#FFFFFF transparent transparent transparent",
    font: "Acumin Pro, Arial, sans-serif !important;"
};