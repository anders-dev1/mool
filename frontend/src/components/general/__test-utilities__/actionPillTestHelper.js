import {screen} from "@testing-library/react";

export function findActionPillByIcon(icon) {
    return screen.getByText(icon).closest('div');
}

export function findActionPillNumberByIcon(icon) {
    const pill = findActionPillByIcon(icon);
    const numberP = pill.lastChild.lastChild;
    return numberP.textContent;
}