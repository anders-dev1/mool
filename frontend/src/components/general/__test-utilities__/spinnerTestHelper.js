export function findSpinner(container){
    return container.getElementsByClassName('lds-ring')[0];
}