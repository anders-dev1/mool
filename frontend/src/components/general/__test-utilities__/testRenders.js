import React from 'react'
import {render} from '@testing-library/react'
import {Provider} from 'react-redux'
import {setupStore, store} from "../../../redux/store";
import {testTheme} from "./TestTheme";
import {ThemeProvider} from "styled-components";

// This is used to set up the store for tests.
export function renderWithProviders(
    ui,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        testStore = setupStore(preloadedState),
        ...renderOptions
    } = {}
) {
    function Wrapper({children}) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return {testStore, ...render(ui, {wrapper: Wrapper, ...renderOptions})}
}

// This is used with the react-test-renderer to render dom in tests.
export function internalsForRenderer(
    ui,
    {
        preloadedState = {},
        testStore = setupStore(preloadedState)
    } = {}
){
    function Wrapper(children) {
        return(
            <Provider store={testStore}>
                <ThemeProvider theme={testTheme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    }

    return Wrapper(ui);
}