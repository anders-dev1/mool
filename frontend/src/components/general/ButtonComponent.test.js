import Button from "./ButtonComponent";
import {fireEvent, render, screen} from "@testing-library/react";
import {testTheme} from "./__test-utilities__/TestTheme";
import {ThemeProvider} from "styled-components";
import {MemoryRouter, Route} from "react-router-dom";
import renderer from "react-test-renderer";

test('renders correctly', () => {
    const domTree = renderer
        .create(
            <ThemeProvider theme={testTheme}>
                <Button class={"classname"} meaty primary/>
            </ThemeProvider>
        ).toJSON()
    expect(domTree).toMatchSnapshot();
});

test('when provided processing value is true, spinner should show', () => {
    // Arrange
    render(
        <ThemeProvider theme={testTheme}>
            <Button processing/>
        </ThemeProvider>
    );

    // Assert
    const component = screen.getByTestId('spinner');
    expect(component).toHaveStyle('display: flex');
});

describe('button clicked', () => {
    test('when onClick callback is provided, should activate callback', () => {
        // Arrange
        const mockCallBack = jest.fn();
        const {container} = render(
            <ThemeProvider theme={testTheme}>
                <Button onClick={mockCallBack}/>
            </ThemeProvider>);
        const component = container.querySelector('button');

        // Act
        fireEvent.click(component);

        // Assert
        expect(mockCallBack).toHaveBeenCalled();
    });

    test('when link is provided, should route to link', () => {
        // Arrange
        const testPath = "test";
        let testLocation;

        const {container} = render(
            <ThemeProvider theme={testTheme}>
                <MemoryRouter>
                    <Button link={testPath}/>
                    <Route
                        path="*"
                        render={({location}) => {
                            testLocation = location;
                            return null;
                        }}/>
                </MemoryRouter>
            </ThemeProvider>);
        const component = container.querySelector('button');

        // Act
        fireEvent.click(component);

        // Assert
        expect(testLocation.pathname).toEqual(`/${testPath}`);
    });
});