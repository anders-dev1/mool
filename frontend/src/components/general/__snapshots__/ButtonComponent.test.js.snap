// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-hKMtZM ivzQBy"
>
  <div
    className="sc-eCYdqJ cWdtZE"
    data-testid="childcontainer"
  />
  <div
    className="sc-jSMfEi cfugSa"
    data-testid="spinner"
  >
    <div
      className="sc-bczRLJ jdNppJ lds-ring"
      size="1.6"
    >
      <div
        className="sc-gsnTZi BOYSF"
        size="1.6"
      />
      <div
        className="sc-gsnTZi BOYSF"
        size="1.6"
      />
      <div
        className="sc-gsnTZi BOYSF"
        size="1.6"
      />
      <div
        className="sc-gsnTZi BOYSF"
        size="1.6"
      />
    </div>
  </div>
</button>
`;
